### DISCUSSION TITLE
authenticated EOF block

### DISCUSSION THREAD

**Discussion by overheadhunter** [encryption-alliance/unified-vault-format#14] [Approved]
Let `n` be the number of bytes per block (see #13).

The last block of a file MUST NOT contain `n` bytes. If the cleartext happens to be a multiple of `n`, a zero-byte-block (just nonce + tag, block number mixed in as AAD as per #8) MUST be appended.

This protects against unnoticed tampering of the EOF.

**Comment by ncw** [comment]
Could a zero byte block and a block where everything but the header was truncated be discriminated?

**Reply by rfjakob** [reply]
Yes, should, because with authenticated encryption you usually have

    [nonce][user data][message authentication code]

and tampering with anything means decryption will fail.

Cutting off after [nonce] would fail with a parse error, I guess.

**Reply by ncw** [reply]
Ah yes that sounds obvious now you say it :-)
